DigitalLearningSolutions.Web/Scripts/searchSortFilterAndPaginate/searchSortFilterAndPaginate.ts

```diff
@@ -290,16 +299,19 @@ export class SearchSortFilterAndPaginate {
 
     const pageNumber = this.page ?? currentPageNumber;
     const queryParametersToRetain = this.getQueryParametersForUpdatedURL();
-    const newUrl = `${urlParts.join('/')}/${pageNumber}${queryParametersToRetain}`;
-    window.history.replaceState({}, '', newUrl);
+
+    const returnId = window.location.hash;
+    if (returnId) {
+      const newUrl = `${urlParts.join('/')}/${pageNumber}${queryParametersToRetain}${returnId}`;
+      window.history.replaceState({}, '', newUrl);
+    } else {
+      const newUrl = `${urlParts.join('/')}/${pageNumber}${queryParametersToRetain}`;
+      window.history.replaceState({}, '', newUrl);
+    }

```

Comment:
Yeah, this was keeping the fragment. I've changed the logic here so that it only preserves the fragment on initial page load, but not when we interact with the controls